File: 2478-longest-nice-subarray/longest-nice-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestNiceSubarray = function (nums) {
    let max = 1
    let left = 0
    let used = 0
    for (let i = 0; i < nums.length ; i++) {
        while ((used & nums[i]) != 0) {
            used ^= nums[left]
            left++
        }
        used |= nums[i]
        max = Math.max(max, i - left + 1)
    }
    return max
};

File: 2478-longest-nice-subarray/longest-nice-subarray.py
class Solution:
    def longestNiceSubarray(self, nums: List[int]) -> int:
        max_value = 1
        left = 0
        used = 0
        for i in range(len(nums)):
            while (used & nums[i]) != 0:
                used ^= nums[left]
                left += 1
            used |= nums[i]
            max_value = max(max_value, i - left + 1)
        return max_value 